### ISSUE TITLE
Taking away space when pasting SIRET number into Client

### ISSUE THREAD

**Issue by hasse-basse** [Dolibarr/dolibarr#29241]
### Feature Request

In the French version of 19.0.1,  Numéro de SIRET is used for clients. I either have it from my clients or I find it on Infogreffe or similar sites and it is almost always written in the format "123 123 123 12345", but when I copy and paste it into the field for the SIRET number in /htdocs/societe/card.php  the result will be "123 123 123 12" so I have to edit it manually.

I think it must be possible to add a function that takes away the space, or? It would be very helpfull.

**Comment by JonBendtsen** [comment]
this field is probably used in other countries which may or may not require spaces.

This page writes the format with spaces and even letters.
https://www.startbusinessinfrance.com/blog/post/tip-what-is-a-siren-or-siret-number
`RCS PARIS 453 983 245`
`RC NANTES 234 987 456`

What if the field is just made longer? Wouldn't that solve your problems?

**Comment by hasse-basse** [comment]
That is not the same, France is confusing. there are 3 different systems in place, in parallel. What I talk about is Numéro de SIRET, and it is only numbers, and it is 123 123 123 12345

- Id. prof. 1 (SIREN)	123 123 123 (this is the company ID without telling the address. 9 figures and no letters
- Id. prof. 2 (SIRET)     123 123 123 12345 (this one is based on the on above but the last 5 figures is unique for each address of the company)
- Id. prof. 3 (NAF-APE)	This one tell what kind of activity it is about Letters and figures
- Id. prof. 4 (RCS/RM)	This is the one you brought up as an example. It is on the way out, but is still around. It is the same as SIREN but also indicates where the registration has been done (Paris, Nantes ...)	
- Numéro de TVA : Vat number. The same as SIREN but FRxx in the beginning

The field for  the Numéro de SIREN does automatically take away the space when you paste into the filed, but the field for the SIRET does not. There is a function in PHP that can do this, it is just to add it, but I don't know how. The Numéro de SIRET is always 14 figures long and there are never letters and we should not change this, because then errors is more likely to happen.

**Comment by JonBendtsen** [comment]
@hasse-basse  can you provide a picture of where it is shown inside dolibarr?

**Comment by hasse-basse** [comment]
In

/htdocs/societe/card.php

when you create a company situated in France. So just create a new client and set its address to France.

**Comment by JonBendtsen** [comment]
> /htdocs/fourn/commande/card.php

vendor order? how is that a client?

**Comment by hasse-basse** [comment]
Sorry, 

/htdocs/societe/card.php

I had multiple windows open. Just create a new client with the location in France, and you will see.

**Comment by JonBendtsen** [comment]
> /htdocs/societe/card.php

I see it now. The "professional id" fields that I've always wondered about what usage they had.

They are a varchar(128) in my database, so they should be big enough to hold a much longer text. I wonder if Dolibarr GUI cuts them off?

![professional_id](https://github.com/Dolibarr/dolibarr/assets/2551481/dd2a3b51-b566-473d-8335-21ef219f8346)

**Comment by JonBendtsen** [comment]
@hasse-basse I can't find any length cut off, but there is this trim

```
				$object->idprof1				= GETPOST('idprof1', 'alphanohtml');
				$object->idprof2				= GETPOST('idprof2', 'alphanohtml');
				$object->idprof3				= GETPOST('idprof3', 'alphanohtml');
				$object->idprof4				= GETPOST('idprof4', 'alphanohtml');
				$object->idprof5				= GETPOST('idprof5', 'alphanohtml');
				$object->idprof6				= GETPOST('idprof6', 'alphanohtml'); 
```

**Comment by hasse-basse** [comment]
We don't want it longer. It should be 123 123 123 12345 = 14 figures. Noting more, nothing less to lower the risk for mistakes. The French SIRET number is always 14 figures long. So what should be done in the PHP control of the field is to rinse all space if it  pasted into the field.

**Comment by JonBendtsen** [comment]
> We don't want it longer. It should be 123 123 123 12345 = 14 figures. Noting more, nothing less to lower the risk for mistakes. The French SIRET number is always 14 figures long. So what should be done in the PHP control of the field is to rinse all space if it pasted into the field.

but it is a different field in different countries. Else where it is called idprod1 to idprof6. And it is 128 chars long in the database. So there must be something in Dolibarrs PHP code that limits the length

**Comment by hasse-basse** [comment]
Yes. It is in the code, it must be and I know this can be controlled in the code, I just don't know how to do it. It is not a DB issue

**Comment by JonBendtsen** [comment]
> Yes. It is in the code, it must be and I know this can be controlled in the code, I just don't know how to do it. It is not a DB issue

I wonder if it is some javascript that does it?

**Comment by JonBendtsen** [comment]
I think it is either Javascript or somewhere in the PHP I haven't found due to it works fine in the API with values longer than what the GUI accept.

![dolibarr_profid](https://github.com/Dolibarr/dolibarr/assets/2551481/31d8f22b-42a6-48e1-9cbd-6ec949130711)

![json_profid](https://github.com/Dolibarr/dolibarr/assets/2551481/7bd43ab4-2f9a-4de6-93a3-4f9c80a40c64)

![dolibarr_after_API_PUT_idprof](https://github.com/Dolibarr/dolibarr/assets/2551481/016351cc-9d80-47a3-b059-0a627f75feb9)

**Comment by hasse-basse** [comment]
I'm not a coder, but I've seen PHP doing this. I found this when searching

https://stackoverflow.com/questions/55691297/how-to-remove-space-in-input-data

**Comment by JonBendtsen** [comment]
@hasse-basse  but that is not what I am looking for

**Comment by hasse-basse** [comment]
I don't know, as said I'm not a programmer and just hope that someone can solve this. It is not a huge issue but it is annoying.
